modular_skyrat/modules/customization/modules/clothing/~donator/donator_items.dm

```diff
@@ -19,3 +19,40 @@
 	icon_state = "khi_cig_packet"
 	base_icon_state = "khi_cig_packet"
 	spawn_type = /obj/item/clothing/mask/cigarette/khi
+
+//Donator reward for Stonetear
+/obj/item/hairbrush/switchblade
+	name = "switchcomb"
+	icon = 'modular_skyrat/master_files/icons/donator/obj/custom.dmi'
+	icon_state = "switchblade"
+	base_icon_state = "switchblade"
+	desc = "A sharp, concealable, spring-loaded comb."
+	hitsound = 'sound/weapons/genhit.ogg'
+	resistance_flags = FIRE_PROOF
+	var/extended = FALSE
+
+/obj/item/hairbrush/switchblade/Initialize(mapload)
+	. = ..()
+	AddElement(/datum/element/update_icon_updates_onmob, ITEM_SLOT_HANDS)
+
+//This is called when you transform it
+/obj/item/hairbrush/switchblade/attack_self(mob/user, modifiers)
+	if(src.extended)
+		src.extended = FALSE
+		icon_state = "switchblade"
+	else
+		icon_state = "switchblade_on"
+		src.extended = TRUE
+	playsound(user ? user : src, 'sound/weapons/batonextend.ogg', 30, TRUE)
+
+
+// This makes it so you have to extend it.
+/obj/item/hairbrush/switchblade/attack(mob/target, mob/user)
+	if(src.extended == FALSE)
+		to_chat(usr, span_warning("Try extending the blade first, silly!"))
+		return
+	if(target.stat == DEAD)
```

Comment:
Just call the parent's attack instead of presumably duping comb code

Reply:
I'm using this to check if the comb is actually extended.